src/services/transactionHistory.ts

```diff
@@ -201,6 +175,60 @@ export const askTransactionHistory = async (
 
 };
 
+const rowToCertificate = (row:any):Certificate => {
+  switch(row.jsType){
+  case "StakeRegistration":
+    return { kind: row.jsType
+      , stakeCredential: row.stakeCred };
+  case "StakeDeregistration":
+    return { kind: row.jsType
+      , stakeCredential: row.stakeCred };
+  case "StakeDelegation":
+    return { kind: row.jsType
+      , poolKeyHash: row.poolHashKey
+      , stakeCredential: row.stakeCred };
+  case "PoolRegistration": {
+    const poolRelays = row.poolParamsRelays 
+      ? row.poolParamsRelays.map((obj:any) => ({
+        ipv4: obj.ipv4
+        , ipv6: obj.ipv6
+        , dnsName: obj.dnsName
+        , dnsSrvName: obj.dnsSrvName
+        , port: obj.port
+      }))
+      : [];
+                                                     
+    const params = { 
+      operator: row.poolParamsOperator
+      , vrfKeyHash: row.poolParamsVrfKeyHash
+      , pledge: row.poolParamsPledge
+      , cost: row.poolParamsCost
+      , margin: row.poolParamsMargin
+      , rewardAccount: row.poolParamsRewardAccount
+      , poolOwners: row.poolParamsOwners
+      , relays: poolRelays
+      , poolMetadata: row.poolParamsMetaDataUrl === null 
+        ? null
+        : { url: row.poolParamsMetaDataUrl
+          , metadataHash: row.poolParamsMetaDataHash }
+    };
+    return { kind: row.jsType
+      , poolParams: params};
+  }
+  case "PoolRetirement":
+    return { kind: row.jsType
+      , poolKeyHash: row.poolHashKey
+      , epoch: row.epoch };
+  case "MoveInstantaneousRewardsCert":
+    return { kind: row.jsType
+      , pot: row.mirPot
+      , rewards: row.rewards === null
+        ? []
+        : row.rewards.map( (o:any)=> o.f1) };
+  default:
+    throw Error(`Certificate from DB doesn't match any known type: ${row}`);
```

Comment:
I would log instead of throwing an error here. Otherwise, if IOHK adds a new certificate type in the future and we don't update our server, things will start crashing

Reply:
the server does not crash, the app already is set to log all errors.  it will return a 500, however.  I'll allow it to return a 200 with the passing certs.